Conceptpower+Spring/src/test/java/edu/asu/conceptpower/rest/ConceptSearchTest.java

```diff
@@ -0,0 +1,268 @@
+                Mockito.anyString(), Mockito.eq(1), Mockito.eq(numberOfRecordsPerPage));
+
+        RestTestUtility.testValidJson(output.getBody());
+        Assert.assertEquals(expectedOutput, output.getBody());
+        Assert.assertEquals(HttpStatus.OK, output.getStatusCode());
+
+    }
+
+    @Test
+    public void test_searchConcept_noRecordsFoundInJSON() throws IllegalAccessException, LuceneException,
+            IndexerRunningException, IllegalArgumentException, JSONException, IOException {
+
+        final String jsonNoRecordsFoundError = IOUtil
+                .toString(this.getClass().getClassLoader().getResourceAsStream("output/recordsNotFound.json"));
+        Mockito.when(result.hasErrors()).thenReturn(false);
+        Map<String, String> map = new HashMap<>();
+        map.put("pos", pos);
+        map.put("type-id", typeId);
+        map.put("concept_list", conceptListName);
+        map.put("word", word);
+
+        int numberOfResults = 100;
+
+        Mockito.when(manager.getTotalNumberOfRecordsForSearch(Mockito.anyMapOf(String.class, String.class),
+                Mockito.anyString())).thenReturn(numberOfResults);
+
+        Mockito.when(manager.searchForConceptByPageNumberAndFieldMap(Mockito.anyMapOf(String.class, String.class),
+                Mockito.anyString(), Mockito.eq(1), Mockito.eq(numberOfRecordsPerPage))).thenReturn(null);
+
+        ResponseEntity<String> output = conceptSearch.searchConcept(conceptSearchParameter, result,
+                MediaType.APPLICATION_JSON_VALUE);
+
+        Mockito.verify(manager).getTotalNumberOfRecordsForSearch(Mockito.anyMapOf(String.class, String.class),
+                Mockito.anyString());
+
+        Mockito.verify(manager).searchForConceptByPageNumberAndFieldMap(Mockito.anyMapOf(String.class, String.class),
+                Mockito.anyString(), Mockito.eq(1), Mockito.eq(numberOfRecordsPerPage));
+
+        Assert.assertEquals(HttpStatus.OK, output.getStatusCode());
+        Assert.assertEquals(jsonNoRecordsFoundError, output.getBody());
+    }
+
+    @Test
+    public void test_searchConcept_noRecordsFoundInXML() throws IllegalAccessException, LuceneException,
+            IndexerRunningException, IllegalArgumentException, JSONException, IOException {
+
+        final String xmlNoRecordsFoundError = IOUtil
+                .toString(this.getClass().getClassLoader().getResourceAsStream("output/recordsNotFound.xml"));
+
+        Mockito.when(result.hasErrors()).thenReturn(false);
+        Map<String, String> map = new HashMap<>();
+        map.put("pos", pos);
+        map.put("type-id", typeId);
+        map.put("concept_list", conceptListName);
+        map.put("word", word);
+
+        int numberOfResults = 100;
+
+        Mockito.when(manager.getTotalNumberOfRecordsForSearch(Mockito.anyMapOf(String.class, String.class),
+                Mockito.anyString())).thenReturn(numberOfResults);
```

Comment:
I changed this one to 2. And when there are no records found I made this one to 0